component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdPrincipalProcessor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2012 eXo Platform SAS.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.gatein.security.oauth.openid;
+
+import org.gatein.security.oauth.spi.OAuthPrincipal;
+import org.gatein.security.oauth.spi.OAuthPrincipalProcessor;
+import org.gatein.security.oauth.utils.OAuthUtils;
+
+import org.exoplatform.services.organization.User;
+import org.exoplatform.services.organization.impl.UserImpl;
+
+public class OpenIdPrincipalProcessor implements OAuthPrincipalProcessor {
+    @Override
+    public User convertToGateInUser(OAuthPrincipal principal) {
+        String email = principal.getEmail();
+        String username = principal.getUserName();
+        if(email != null) {
+            int index = email.indexOf('@');
+            if(index > 0) {
+                username = email.substring(0, index);
+            }
+        }
```

Comment:
I think we do not need to change th eusername unless it exists or it is not available in the OAuthPrincipal
If the username is not found, we may use the firstname + "." + lastname and avoid using the email username, WDYT ?

Reply:
In fact, this is call when the user successfully logs in against his identityProvider, but not exists in exo.
In this case, which username should we use ? 
I decide to take the first part of the email. 
In fact, this login will be never use by the user as he will always logs with his email with the IdentityProvider.

Reply:
An OpenID authenticated user could login with his username & password if he wishes. He needs just to reset his password, but he won't know what username was assigned to him. That is why I thought that his first+lastname could be a good option WDYT ?

Reply:
Yes, he can, but once he have used and oauth system to first login and create his account, dont you think he will still use the oauth mecanism ? In addition, if he really want to login directly in exo, I think he will use his email, has he can now.